### PR TITLE
Adding virtual mode validation for a double click

### LINKED ISSUE #4156: Double click allows to check item in the ListView that is in virtual mode
.NET Core Version: 
- .Net SDK 5.0.100-rc.2.20479.15
- .Net SDK 3.1.7

Have you experienced this same bug with .NET Framework?: 
- Yes

**Minimal repro**
1. Create application with textbox and ListView that is in virtual mode
2. Enable Checkboxes and add items to ListView.
3. Or use attached project ([ListViewVirtualMode.zip](https://github.com/dotnet/winforms/files/5438099/ListViewVirtualMode.zip))
4. Run application
5. Execute double click on checkbox of any item
6. Switch the focus to the textbox and move the focus back to the ListView.
listview-doubleclick

**Actual behavior:**
Checkbox changes state

**Expected behavior:**
Checkbox should not change state.

**Note:**
We have additional validation of virtual mode for [mouse click](https://github.com/dotnet/winforms/blob/26ee7b5fea05516d1caa83b94995c54115c148a1/src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs#L5813) and ["space" key](https://github.com/dotnet/winforms/blob/26ee7b5fea05516d1caa83b94995c54115c148a1/src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs#L6587). Looks like we forgot to add a similar validation for double click

### PR DESCRIPTION
**Pull request by SergeySmirnov-Akvelon**
Fixes #4156

## Proposed changes
- Added virtual mode validation for a double click

## Customer Impact
The checkbox will not change its value on a double click when ListView is in virtual mode.
Before fix:
Issue-4290-beforefix

After fix:
Issue-4290-afterfix

## Regression? 
- No

## Risk
- Minimal

## Test methodology 
- Manually

## Test environment(s) 
- Microsoft Windows [Version 10.0.19041.388]
- .Net SDK 5.0.100-rc.2.20479.15

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/4290)

### CODE REVIEW

**Review by RussKie — DISMISSED** [review]

**Review by RussKie — COMMENTED** [review]
Can we add a test?

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs [hunk 1]
```diff
@@ -72,12 +73,82 @@ public ScenarioResult Click_On_Second_Column_Does_Not_Alter_Checkboxes(TParams p
...
+                if (setOldCursorPos.IsTrue())
+                {
+                    // Move cursor to old position
+                    MouseHelper.ChangeMousePosition(previousPosition.X, previousPosition.Y);
+                    Application.DoEvents();
+                }
+            }
+        }
+
+        private void InitializeItems(ListView listView, bool virtualModeEnabled, bool CheckBoxesEnabled, TParams p)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs:149]: argument name should be camelCased. CheckBoxesEbnabled-> checkBoxesEnabled

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs [hunk 2]
```diff
@@ -32,7 +33,7 @@ public static void Main(string[] args)
         [Scenario(true)]
         public ScenarioResult Click_On_Second_Column_Does_Not_Alter_Checkboxes(TParams p)
         {
-            InitializeItems(_listView, p);
+            InitializeItems(_listView, false, true, p);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs:36]: Consider adding argument names in this and other invocations of `InitializeItems`.

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs [hunk 3]
```diff
@@ -87,7 +158,30 @@ private void InitializeItems(ListView listView, TParams p)
...
+                {
+                    0 => listViewItem1,
+                    1 => listViewItem2,
+                    2 => listViewItem3,
+                    _ => listViewItem1,
+                };
+            };
+
+            listView.Items.Clear();
+            Thread.Sleep(100);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs:174]: Why is this delay needed?

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiTestsHelper/MouseHelper.cs [hunk 4]
```diff
@@ -32,6 +37,19 @@ public static void SendClick(int x, int y)
             }
         }
 
+        public static void SendDoubleClick(int x, int y)
+        {
+            bool mouseSwapped = GetSystemMetrics(SystemMetric.SM_SWAPBUTTON) != 0;
+            SendMouseInput(x, y, MOUSEEVENTF.MOVE | MOUSEEVENTF.ABSOLUTE);
+            SendMouseInput(0, 0, mouseSwapped ? MOUSEEVENTF.RIGHTDOWN : MOUSEEVENTF.LEFTDOWN);
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiTestsHelper/MouseHelper.cs:44]: Consider adding argument name before the integer constant.

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good. @vladimir-krestov - could you please review this PR if you had not don so already?

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs (hunk 3 above)
Comment (SergeySmirnov-Akvelon) [reply src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiListViewTests/MauiListViewTests.cs]: Thank you. This is unnecessary code left over after investigation

**Review by RussKie — APPROVED** [review]

**Pull request merged**